From 8922130bc08f311ad00e40e3dfe0fc05f959b0d9 Mon Sep 17 00:00:00 2001
From: Stephen Smalley <stephen.smalley.work@gmail.com>
Date: Tue, 3 Mar 2026 15:42:40 -0500
Subject: [PATCH 87/90] selinuxfs: allow mounting in a user namespace

This is required to mount the selinuxfs instance for a SELinux namespace
in a non-init user namespace. This is safe because a new selinuxfs
superblock is only created per SELinux namespace, not per-userns.
Consequently, selinuxfs can only be mounted in a child userns if
the SELinux namespace is also unshared, at which point a new
selinuxfs instance will be created for that namespace and operations
on it will remain private to the child SELinux namespace.

Signed-off-by: Stephen Smalley <stephen.smalley.work@gmail.com>

Upstream-commit: 287159dc1f77ff17c4c22bd4cd587f14e64fca0f
---
 security/selinux/selinuxfs.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/security/selinux/selinuxfs.c b/security/selinux/selinuxfs.c
index 115f121813e66fc42398034bd38ac3870b0b27cc..00493bf3f20023e6ef42871d7fe82bd101a26a87 100644
--- a/security/selinux/selinuxfs.c
+++ b/security/selinux/selinuxfs.c
@@ -2322,6 +2322,9 @@ static struct file_system_type sel_fs_type = {
 	.name		= "selinuxfs",
 	.init_fs_context = sel_init_fs_context,
 	.kill_sb	= sel_kill_sb,
+#ifdef CONFIG_SECURITY_SELINUX_NS
+	.fs_flags	= FS_USERNS_MOUNT,
+#endif
 };
 
 struct path selinux_null __ro_after_init;
